File: Excel/454 Challenge.py
import pandas as pd
import re

input = pd.read_excel("454 Extraction of number of nodes.xlsx", usecols="A")
test = pd.read_excel("454 Extraction of number of nodes.xlsx", usecols="B")

def replace_notation_with_range(text_vector):
    def replace_match(match):
        numbers = list(map(int, match.group().split(" to ")))
        range_values = list(range(numbers[0], numbers[1]+1))
        return ", ".join(map(str, range_values))
    
    return [re.sub("\\d+ to \\d+", replace_match, text) for text in text_vector]

def count_numbers(text_vector):
    return [len(re.findall("\\d+", text)) for text in text_vector]

result = input.copy()
result["Pronlem"] = result["Pronlem"].str.lower()
result["Pronlem"] = replace_notation_with_range(result["Pronlem"])
result["Count"] = count_numbers(result["Pronlem"])
result = result[["Count"]]

print(result["Count"].equals(test["Answer Expected"])) # True

File: Excel/400-499/454/454 Challenge.R
library(tidyverse)
library(readxl)

input = read_excel("Excel/454 Extraction of number of nodes.xlsx", range = "A1:A9")
test  = read_excel("Excel/454 Extraction of number of nodes.xlsx", range = "B1:B9")

replace_notation_with_range <- function(text_vector) {
  str_replace_all(text_vector, "\\d+ to \\d+", function(match) {
    numbers <- str_split(match, " to ") %>%
      unlist() %>%
      as.numeric()
    
    range <- seq(from = numbers[1], to = numbers[2])
    paste(range, collapse = ", ")
  })
}

count_numbers <- function(text_vector) {
  str_count(text_vector, "\\d+") %>%
    as.numeric()
}

result = input %>%
  mutate(Pronlem = str_to_lower(Pronlem)) %>%
  mutate(Pronlem = map_chr(Pronlem, replace_notation_with_range)) %>%
  mutate(Count = count_numbers(Pronlem)) %>%
  select(Count)

identical(result$Count, test$`Answer Expected`)
# [1] TRUE